crates/mdbook-mun-examples-test/src/main.rs

```diff
@@ -0,0 +1,120 @@
+use mdbook::renderer::RenderContext;
+use mdbook::BookItem;
+
+use std::io;
+
+use regex::Regex;
+
+use mun_test::{CompileAndRunTestDriver, CompileTestDriver};
+
+use mun_runtime::RuntimeBuilder;
+
+use std::panic;
+
+enum TestingMethod {
+    Skip,
+    CompileFail,
+    CompileAndRunFail,
+    Compile,
+    CompileAndRun, // default
+}
+
+fn get_testing_method(code: &str) -> TestingMethod {
+    let mut testing_method = TestingMethod::CompileAndRun;
+    for line in code.lines() {
+        if let Some(first_char) = line.chars().next() {
+            if first_char != ',' {
+                return testing_method;
+            }
+        } else {
+            return testing_method;
+        }
+        if line == ",compile_fail" {
+            match testing_method {
+                TestingMethod::Compile => testing_method = TestingMethod::CompileFail,
+                TestingMethod::CompileAndRun => testing_method = TestingMethod::CompileAndRunFail,
+                _ => (),
+            }
+        }
+        if line == ",no_run" {
+            match testing_method {
+                TestingMethod::CompileAndRun => testing_method = TestingMethod::Compile,
+                TestingMethod::CompileAndRunFail => testing_method = TestingMethod::CompileFail,
+                _ => (),
+            }
+        }
+        if line == ",skip" {
+            return TestingMethod::Skip;
+        }
+    }
+
+    testing_method
+}
+
+fn config_fn(runtime_builder: RuntimeBuilder) -> RuntimeBuilder {
+    runtime_builder
+}
+
+fn test_code(code: &str) {
+    let testing_method = get_testing_method(code);
+
+    // Removing '#' from code
+    let mut code = code.replace("\n#", "\n");
+
+    // Removing testing commands like "no_run", "compile_fail"
+    for testing_command in ["no_run", "compile_fail"].iter() {
+        code = code.replace(format!(",{}", testing_command).as_str(), "");
+    }
+
+    match testing_method {
+        TestingMethod::CompileFail => {
+            let previous_hook = panic::take_hook();
+            panic::set_hook(Box::new(|_| {}));
+
+            if panic::catch_unwind(|| CompileTestDriver::new(&code)).is_ok() {
+                panic::set_hook(previous_hook);
+                panic!("Code that should have caused the error compiled successfully ❌")
+            }
+
+            panic::set_hook(previous_hook);
+        }
+        TestingMethod::CompileAndRunFail => {
+            let previous_hook = panic::take_hook();
+            panic::set_hook(Box::new(|_| {}));
+
+            if panic::catch_unwind(|| CompileAndRunTestDriver::new(&code, config_fn).unwrap())
```

Comment:
Instead of catching panics (which should never happen), couldn't you catch the error return code here? (WDYT @Wodann ?)

Reply:
Could catch, instead of panicking on `unwrap`, you can just directly check `.is_ok()` on the result of the `new()` call.

Reply:
@baszalmstra @Wodann i tested, i can't just use here `.is_ok()`